### DISCUSSION TITLE
how would you go about integrating altair into idom?

### DISCUSSION THREAD

**Discussion by acivitillo** [idom-team/idom#619] [Question] [answered]
Altair: https://altair-viz.github.io/

I tried but failed here. There is a an option to save Altair charts in HTML so I guess I could embed them into idom, but I was wondering if there was a better way?

**Comment by rmorshea** [comment, marked as answer]
IDOM provides an interface for [loading custom javascript](https://idom-docs.herokuapp.com/docs/escape-hatches/javascript-components.html#custom-javascript-components) and a [template repository](https://github.com/idom-team/idom-react-component-cookiecutter) that should help you get set up.

I'd be really curious to get feedback on this area of the docs since, to my knowledge, I'm the only one who's [made a custom component](https://github.com/idom-team/idom-plotly).

**Reply by sylvoslee** [reply]
How to install material ui locally, because my company's server doesn't have internet access

**Reply by rmorshea** [reply]
You'll need to build a Javascript bundle containing Material UI with [webpack](https://webpack.js.org/) or [rollup](https://rollupjs.org/guide/en/) and load it into IDOM with [`module_from_file`](https://idom-docs.herokuapp.com/docs/_auto/apis.html#idom.web.module.module_from_file). Once you go down this route though, it starts to look like a custom component. In which case, you might just want to do that from the start.

**Reply by acivitillo** [reply]
Your idom-plotly answers my question pretty much
